// @gettruck/payfast-js/lib/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var buildUrl = require("build-url");
var PayFastAPI = /** @class */ (function () {
    function PayFastAPI(_a) {
        var merchant_id = _a.merchant_id, merchant_key = _a.merchant_key, production = _a.production;
        this.merchant_id = merchant_id;
        this.merchant_key = merchant_key;
        this.url = "https://" + (production ? "payfast" : "sandbox.payfast") + ".co.za/eng/process";
        this.payment_details = {
            amount: null,
            item_name: null
        };
        this.reference_details = {
            m_payment_id: null,
            custom_str1: null,
            custom_str2: null
        };
        this.redirect_urls = {};
    }
    PayFastAPI.prototype.addPaymentDetails = function (payment_details) {
        this.payment_details.amount = payment_details.amount;
        this.payment_details.item_name = payment_details.item_name;
        this.payment_details.item_description = payment_details.item_description;
        this.payment_details.currency = payment_details.currency;
        this.payment_details.payment_method = payment_details.payment_method;
        this.payment_details.name_first = payment_details.name_first;
        this.payment_details.name_last = payment_details.name_last;
        this.payment_details.email_address = payment_details.email_address;
        this.payment_details.cell_number = payment_details.cell_number;
    };
    PayFastAPI.prototype.addReferenceDetails = function (reference_details) {
        this.reference_details.m_payment_id = reference_details.m_payment_id;
        this.reference_details.custom_str1 = reference_details.custom_str1;
        this.reference_details.custom_str2 = reference_details.custom_str2;
    };
    PayFastAPI.prototype.cancelURL = function (url) {
        this.redirect_urls.cancel_url = url;
    };
    PayFastAPI.prototype.returnURL = function (url) {
        this.redirect_urls.return_url = url;
    };
    PayFastAPI.prototype.notifyURL = function (url) {
        this.redirect_urls.notify_url = url;
    };
    PayFastAPI.prototype.generateURL = function () {
        // @ts-ignore
        return buildUrl(this.url, {
            queryParams: __assign({ merchant_id: this.merchant_id, merchant_key: this.merchant_key }, this.payment_details, this.redirect_urls, this.reference_details)
        });
    };
    return PayFastAPI;
}());
exports.PayFastAPI = PayFastAPI;
