// @gettruck/payfast-js/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const buildUrl = require("build-url");
class PayFastAPI {
    constructor({ merchant_id, merchant_key }) {
        this.merchant_id = merchant_id;
        this.merchant_key = merchant_key;
        this.url = "https://sandbox.payfast.co.za/eng/process";
        this.payment_details = {
            amount: null,
            item_name: null
        };
        this.reference_details = {
            m_payment_id: null,
            custom_str1: null,
            custom_str2: null
        };
        this.redirect_urls = {};
    }
    addPaymentDetails(payment_details) {
        this.payment_details.amount = payment_details.amount;
        this.payment_details.item_name = payment_details.item_name;
        this.payment_details.item_description = payment_details.item_description;
        this.payment_details.currency = payment_details.currency;
        this.payment_details.payment_method = payment_details.payment_method;
        this.payment_details.name_first = payment_details.name_first;
        this.payment_details.name_last = payment_details.name_last;
        this.payment_details.email_address = payment_details.email_address;
        this.payment_details.cell_number = payment_details.cell_number;
    }
    addReferenceDetails(reference_details) {
        this.reference_details.m_payment_id = reference_details.m_payment_id;
        this.reference_details.custom_str1 = reference_details.custom_str1;
        this.reference_details.custom_str2 = reference_details.custom_str2;
    }
    cancelURL(url) {
        this.redirect_urls.cancel_url = url;
    }
    returnURL(url) {
        this.redirect_urls.return_url = url;
    }
    notifyURL(url) {
        this.redirect_urls.notify_url = url;
    }
    generateURL() {
        // @ts-ignore
        return buildUrl(this.url, {
            queryParams: Object.assign({ merchant_id: this.merchant_id, merchant_key: this.merchant_key }, this.payment_details, this.redirect_urls, this.reference_details)
        });
    }
}
exports.default = PayFastAPI;
